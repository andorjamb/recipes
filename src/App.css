.App {
  max-width: var(--app-max-width);
 margin:auto;
}