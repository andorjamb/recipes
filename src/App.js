import React, { useEffect, useState } from 'react';
import { Route, Routes, BrowserRouter, useParams } from 'react-router-dom';
import './App.css';
import axios from 'axios';

import Home from './pages/Home.jsx';
import Recipes from './pages/Recipes.jsx';
import NotFound from './pages/NotFound.jsx';
import RecipeSingle from './pages/RecipeSingle.jsx';
import AddRecipe from './pages/AddRecipe';
import Layout from './pages/Layout';

function App() {
  const params = useParams();
  const [countries, setCountries] = useState([]);
  const [recipeData, setRecipeData] = useState([]);


  async function getCountryData() {
    await axios.get(' https://restcountries.com/v2/all?fields=name,flag')
      .then(res => { setCountries(res.data) });
  }

  async function getRecipeData() {
    await axios.get('http://localhost:3000/recipes')
      .then(res => { setRecipeData(res.data) })
      .catch((err) => console.log(err));
  }
  useEffect(() => {

    getCountryData();
    getRecipeData();

    /* const getCountries = async () => {
      await getCountryData().then(res => setCountries(res));
    }

    const getRecipes = async () => {
      await getRecipeData().then(res => setRecipeData(res));
    } */

   /*  getRecipes();
    getCountries(); */
    console.log(countries);
    console.log(recipeData);
  }, []);

  return (
    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route path='/' element={<Layout />}>
            <Route index element={<Home />} />
            <Route path='/recipes' element={<Recipes countries={countries} recipeData={recipeData} />}></Route>
            <Route path='/recipes/:recipesingle' element={<RecipeSingle recipeData={recipeData} />}></Route>
            <Route path='/addrecipe' element={<AddRecipe countries={countries} />}></Route>
            <Route path='*' element={<NotFound />}></Route>
          </Route></Routes></BrowserRouter>

    </div>
  );
}

export default App;
