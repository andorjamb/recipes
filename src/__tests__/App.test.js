import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from '../App';

afterEach(() => {
    cleanup();
})

test('matches snapshot', () => {
    const recipesTree = renderer.create(<App />).toJSON();
    expect(recipesTree).toMatchSnapshot();
})
