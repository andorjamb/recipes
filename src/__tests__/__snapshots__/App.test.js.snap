// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="layout"
  >
    <header>
      <nav
        className="nav"
      >
        <div
          className="title"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            TasteIT
          </a>
        </div>
        <div
          className="menu"
        >
          <span
            className="menuIcon"
            onClick={[Function]}
          >
            menu
          </span>
        </div>
        <div>
          <ul
            className={null}
          >
            <li>
              <a
                aria-current="page"
                className="active"
                href="/"
                onClick={[Function]}
              >
                Home
              </a>
            </li>
            <li>
              <a
                className=""
                href="/recipes"
                onClick={[Function]}
              >
                Recipes
              </a>
            </li>
            <li>
              <a
                className=""
                href="/addrecipe"
                onClick={[Function]}
              >
                Add Recipe
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </header>
    <div
      className="main"
    >
      <div
        className="home"
      >
        <video
          autoPlay={true}
          className="video"
          data-testid="video"
        >
          <source
            src="Cooking.mp4"
            type="video/mp4"
          />
        </video>
        <div
          className="title"
        >
          <h1
            className="h1"
          >
            TasteIT
          </h1>
        </div>
        <div
          className="menu"
        >
          <div
            className="glassCard"
          >
            <a
              href="/recipes"
              onClick={[Function]}
            >
              <h3>
                Browse Recipes
              </h3>
            </a>
            <p>
              Search for a recipe, or for recipes by country
            </p>
          </div>
          <div
            className="glassCard"
          >
            <a
              href="/addrecipe"
              onClick={[Function]}
            >
              <h3>
                Add your own recipe
              </h3>
            </a>
            <p>
              Add a recipe from your own country to our database
            </p>
          </div>
          <div
            className="glassCard"
          >
            <a
              href="https://en.bc.fi/qualifications/full-stack-web-developer-program/"
              rel="noreferrer"
              target="_blank"
            >
              <h3>
                About HBC
              </h3>
            </a>
            <p>
              This project was completed as part of the Full Stack Developer Program at Helsinki Business College. Find out More!
            </p>
          </div>
        </div>
      </div>
    </div>
    <footer
      className="footer"
    >
      <p>
        Copyright Helsinki Business College 2022-2023
      </p>
      <a
        href="https://github.com/andorjamb/recipes"
      >
        <img
          alt="github"
          className="githubLogo"
          src="github-mark.png"
        />
      </a>
    </footer>
  </div>
   
</div>
`;
